targets/csharp/source/PlayFabSDK/source/Uunit/tests/Utility/JsonUnitTests.cs

```diff
@@ -0,0 +1,414 @@
+            }
+
+            if (failures.Count == 0)
+                testContext.EndTest(UUnitFinishState.PASSED, null);
+            else
+                testContext.EndTest(UUnitFinishState.FAILED, string.Join("\n", failures.ToArray()));
+        }
+
+        [UUnitTest]
+        public void TestDeserializeToObject(UUnitTestContext testContext)
+        {
+            var serializer = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
+            var testInt = serializer.DeserializeObject<object>("1");
+            var testString = serializer.DeserializeObject<object>("\"a string\"");
+            testContext.IntEquals((int)(ulong)testInt, 1);
+            testContext.StringEquals((string)testString, "a string");
+
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        [UUnitTest]
+        public void TestJsonSubObject(UUnitTestContext testContext)
+        {
+            // actualObj contains a real ObjNumFieldTest within subObject
+            var expectedObj = new SerializeJsonSubOjbect { SubObject = new ObjNumFieldTest { ByteValue = 1, DoubleValue = 1, FloatValue = 1, IntValue = 1, LongValue = 1, SbyteValue = 1, ShortValue = 1, UintValue = 1, UlongValue = 1, UshortValue = 1 } };
+            var expectedJson = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer).SerializeObject(expectedObj);
+            // Convert back to SerializeJsonSubOjbect which will serialize the original ObjNumFieldTest to a SimpleJson.JsonObject (or equivalent in another serializer)
+            var actualObj = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer).DeserializeObject<SerializeJsonSubOjbect>(expectedJson);
+            testContext.False(actualObj.SubObject is ObjNumFieldTest, "ObjNumFieldTest should have deserialized as a generic JsonObject");
+            var actualJson = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer).SerializeObject(actualObj);
+            // The real test is that reserializing actualObj should produce identical json
+            testContext.StringEquals(expectedJson, actualJson, actualJson);
+
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        private static bool NullableEquals<T>(T? left, T? right) where T : struct
+        {
+            // If both have a value, return whether the values match.
+            if (left.HasValue && right.HasValue)
+                return left.Value.Equals(right.Value);
+
+            // If neither has a value, return true. If only one does, return false.
+            return !left.HasValue && !right.HasValue;
+        }
+
+        private static bool NullableNotEquals<T>(T? left, T? right) where T : struct
+        {
+            return !NullableEquals(left, right);
+        }
+
+        class StringValueClass
+        {
+            public string StringValue;
```

Comment:
I'm seeing a C# warning declaring this value is never assigned to and will always have its default value null.

You can try making the class internal and set the value to null or even empty string seems to silence the warning.